keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -251,30 +254,79 @@ def call_with_cache(self, token_ids, padding_mask, cache, cache_index):
         )
         return x, cache
 
-    def build_empty_cache(self, batch_size, max_length):
+    def _build_cache(self, prompt):
         """Build an empty cache for use with `call_with_cache()`."""
+        batch_size, max_length = tf.shape(prompt)[0], tf.shape(prompt)[1]
         num_layers = self.backbone.num_layers
         num_heads = self.backbone.num_heads
         head_dim = self.backbone.hidden_dim // self.backbone.num_heads
         shape = [batch_size, num_layers, 2, max_length, num_heads, head_dim]
-        return tf.zeros(shape)
+        cache = tf.zeros(shape)
+        # Seed the cache.
+        _, cache = self.call_with_cache(prompt, cache, 0)
+        return cache
 
-    def _get_token_probability(
+    def compile(
         self,
-        prompt,
-        mask,
-        cache=None,
-        cache_index=None,
+        *args,
+        run_eagerly=False,
+        jit_compile=True,
+        sampler="top_k",
+        **kwargs,
     ):
-        batch_size = tf.shape(prompt)[0]
-        prompt = tf.slice(prompt, [0, cache_index], [batch_size, 1])
-        return self.call_with_cache(prompt, mask, cache, cache_index)
+        xla_compatible = is_xla_compatible(self)
+        super().compile(
+            *args,
+            run_eagerly=run_eagerly,
+            # Only `jit_compile` if not eager and in a compatible environment.
+            jit_compile=jit_compile and xla_compatible and not run_eagerly,
+            **kwargs,
+        )
+        self._sampler = samplers.get(sampler)
+        # Clear the compiled generate function.
+        self.generate_function = None
+
+    def make_generate_function(self):
+        """Create or return the compiled generation function."""
+        # Return the saved generate_function if we have it.
+        if self.generate_function is not None:
+            return self.generate_function
+
+        def generate_function(prompt, input_mask, min_length, max_length):
```

Comment:
I feel `min_length` and `max_length` could be confusing for readers. `min_length` and `max_length` sounds like a pair, but they are from different sources, `min_length` is the minimum length of prompt in the batch, while `max_length` is an arbitrary value. Maybe some comments could help here?

Reply:
Actually we no longer need to pass max_length, so just removed.